src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -240,17 +241,25 @@ private IAnalyzerResult BuildProject(IProjectAnalyzer projectAnalyzer)
             return null;
         }
 
-        private EnvironmentOptions GetEnvironmentOptions(bool requiresNetFramework)
+        private EnvironmentOptions GetEnvironmentOptions(IProjectFile projectFile)
         {
             var os = DetermineOSPlatform();
             EnvironmentOptions options = new EnvironmentOptions();
 
             if (os == OSPlatform.Linux || os == OSPlatform.OSX)
             {
+                var requiresNetFramework = false;
+                try
+                {
+                    requiresNetFramework = projectFile.RequiresNetFramework;
+                }
```

Comment:
I would suggest adding a comment to explain the conditions under which we might expect to catch an error here since the error is thrown in an external dependency (similar explanation as what we discussed this morning would be good).

Reply:
Good suggestion